components/CollectivePickerInviteMenu.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { PlusCircle } from '@styled-icons/feather/PlusCircle';
+
+import { isValidEmail } from '../lib/utils';
+
+import { Box, Flex } from './Grid';
+import StyledButton from './StyledButton';
+import StyledInput from './StyledInput';
+import StyledInputField from './StyledInputField';
+import { Span } from './Text';
+
+export const InviteCollectiveDropdownOption = ({ onClick }) => (
+  <Flex flexDirection="column">
+    <Flex mb="16px">
+      <img width="48px" height="48px" src="/static/images/magnifier.png" />
+      <Box ml="16px">
+        <Span fontSize="12px" fontWeight="700" color="black.800">
+          The person or organization you are looking for is not on Open Collective yet.
+        </Span>
+      </Box>
+    </Flex>
+    <StyledButton borderRadius="14px" onClick={onClick}>
+      <Flex alignItems="center">
+        <PlusCircle size={24} />
+        <Box ml="16px" fontSize="11px">
+          Invite someone to submit an expense
+        </Box>
+      </Flex>
+    </StyledButton>
+  </Flex>
+);
+
+InviteCollectiveDropdownOption.propTypes = {
+  onClick: PropTypes.func.isRequired,
+};
+
+export const InviteCollectiveForm = ({ onCancel, onSave }) => {
+  const [value, setValue] = React.useState({ name: '', email: '', isInvite: true });
+  const setValueProp = prop => e => {
+    e.persist();
+    setValue(v => ({ ...v, [prop]: e.target?.value }));
+  };
+
+  return (
+    <Flex flexDirection="column">
+      <form
+        onSubmit={e => {
+          e.preventDefault();
+          e.stopPropagation();
+          onSave(value);
+        }}
+      >
+        <StyledInputField
+          label="What's the name of who you want to invite?"
```

Comment:
For accessibility, it's important to add `htmlFor` on fields and to pass the `id` provided by `StyledInputField` to the input, It's what links them.